### PR TITLE
Bump version for SqlClient & turn on package build

### PR DESCRIPTION
**Pull request by wtgodbe**
Ensures https://github.com/dotnet/corefx/pull/34260 will ship with 2.2.3 (that PR was a follow-up to https://github.com/dotnet/corefx/pull/31039, which we also shipped out of release/2.2)

@safern @ericstj PTAL

CC @AfsanehR

### CODE REVIEW

**Comment by safern** [comment]
We will also need to update the compat pack on my PR for 2.2.4 -- cc: @danmosemsft

File: pkg/Microsoft.Private.PackageBaseline/packageIndex.json [hunk 1]
```diff
@@ -1274,7 +1274,8 @@
         "4.1.1.0": "4.3.0",
         "4.2.0.0": "4.4.0",
         "4.4.0.0": "4.5.0",
-        "4.5.0.0": "4.6.0"
+        "4.5.0.0": "4.6.0",
+        "4.5.0.1": "4.6.1"
```
Comment (safern) [inline pkg/Microsoft.Private.PackageBaseline/packageIndex.json]: I think you're missing assemblyVersion mapping to package version?

Comment (wtgodbe) [reply pkg/Microsoft.Private.PackageBaseline/packageIndex.json]: I'm confused, isn't that what I'm doing here? 4.6.1 and 4.5.0.1 are the new versions as of this PR

Comment (safern) [reply pkg/Microsoft.Private.PackageBaseline/packageIndex.json]: Oh yeah. Sorry. Misread, thought this where the stable version entry. Disregard

**Comment by wtgodbe** [comment]
Test failures are unrelated, merging

**Pull request merged**